File: Cs/Solution/ConvertDateToBinary.cs
using System;

namespace Solution
{
	public partial class Solution
	{
		public string ConvertDateToBinary(string date)
		{
			string[] parts = date.Split('-');

			string binaryDay = Convert.ToString(int.Parse(parts[0]), 2);
			string binaryMonth = Convert.ToString(int.Parse(parts[1]), 2);
			string binaryYear = Convert.ToString(int.Parse(parts[2]), 2);

			return $"{binaryDay}-{binaryMonth}-{binaryYear}";
		}
	}
}

File: Ruby/solution/convert_date_to_binary.rb
# @param {String} date
# @return {String}
def convert_date_to_binary(date)
  year, month, day = date.split("-").map(&:to_i)
  "#{year.to_s(2)}-#{month.to_s(2)}-#{day.to_s(2)}"
end
